include/boost/dll/detail/demangling/itanium.hpp

```diff
@@ -88,6 +88,108 @@ class mangled_storage_impl : public mangled_storage_base
 
 namespace parser
 {
+    //! declare
+    template <typename... T>
+    struct dummy;
+
+    template <typename T>
+    std::string parse_type_helper(const mangled_storage_impl & ms, dummy<T>*);
+
+    template <typename... T, template <typename...> class Tn>
+    std::string parse_type_helper(const mangled_storage_impl & ms, dummy<Tn<T...>>*);
+
+    template <typename... T, template <typename...> class Tn>
+    std::string parse_type(const mangled_storage_impl & ms, dummy<Tn<T...>>*);
+
+    template <typename T>
+    std::string parse_type(const mangled_storage_impl & ms, dummy<T>*);
+
+    template <typename T1, typename T2, typename... T3>
+    std::string parse_type(const mangled_storage_impl & ms, dummy<T1, T2, T3...>*);
+
+    template <typename R, typename... Args>
+    std::string parse_type(const mangled_storage_impl & ms, dummy<R(Args...)>*);
+
+    std::string parse_type(const mangled_storage_impl & ms, dummy<>*);
+
+    template<typename T>
+    std::string
+    type_name(const mangled_storage_impl &);
+
+    template<typename T>
+    using rm_ref = typename remove_reference<T>::type;
```

Comment:
includes for boost::remove_pointer and boost::remove_reference are missing. Because of that there's an error https://travis-ci.com/boostorg/dll/jobs/264314238#L1138